File: majority_elements.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt = 0;
        int m = 0;

        for (int x : nums) {
            if (cnt == 0) {
                m = x;
                cnt = cnt + 1;
            }
            else {
                if (m == x) {
                    cnt = cnt + 1;
                }
                else {
                    cnt = cnt - 1;
                }
            }
        }

        return m;
    }
};


        

File: Majority Elements.py

#169. Majority Element

class Solution:
     def majorityElement(self, nums: list[int]) -> int:
    
        '''l = len(nums) // 2

        for i in nums:
            if nums.count(i) > l:
                return i'''
   
        d = {}

        for i in nums:
            d[i] = d.get(i, 0) + 1

            if d[i] > len(nums) // 2:
                return i